src/components/Settings/ProfileForm.jsx

```diff
@@ -0,0 +1,224 @@
+import { useState } from "react";
+import Image from "next/image";
+import { useMutation } from "../../lib/apollo";
+import { UPDATE_SETTINGS } from "../../lib/graphql";
+
+const ProfileForm = ({ user }) => {
+  const [loading, setLoading] = useState(false);
+  const [newsletter, setNewsletter] = useState(user.newsletter || false);
+  const [journeyUpdates, setJourneyUpdates] = useState(
+    user.journeyUpdates || false
+  );
+  const [walletAddress, setWalletAddress] = useState(user.connectedWalletAddress || "");
+  const [username, setUsername] = useState(user.username || "");
+
+  const {
+    load: updateSettings,
+    error,
+  } = useMutation(UPDATE_SETTINGS, {
+    onCompleted: (data) => {
+      // TODO - show alert/toast
+      console.log("update setting ", data);
+      setLoading(false);
+    },
+    onError: (error) => {
+      // TODO - show alert/toast
+      console.log("update setting error", error);
+      setLoading(false);
```

Comment:
use "loading" from useMutation hook instead

Reply:
I wasn't sure how to turn loading off once the query completes. Which meant the save button stayed disabled after query complete.

Reply:
It's a state variable and handled by the useMutation, you don't need to handle it will automatically reset once the query finished updating.
So just remove your state "loading" variable and use one from useMutaion like:

const {
    load: updateSettings,
    loading,
    error,
  } = useMutation(...

Reply:
updated :)